{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import dash\n",
    "from dash.dependencies import Input, Output \n",
    "import dash_core_components as dcc\n",
    "import dash_html_components as html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      " * Serving Flask app \"__main__\" (lazy loading)\n",
      " * Environment: production\n",
      "   WARNING: This is a development server. Do not use it in a production deployment.\n",
      "   Use a production WSGI server instead.\n",
      " * Debug mode: off\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " * Running on http://127.0.0.1:8050/ (Press CTRL+C to quit)\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:31] \"\u001b[37mGET / HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:31] \"\u001b[37mGET /_dash-layout HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:31] \"\u001b[37mGET /_dash-dependencies HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:31] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:39] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:40] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:42] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:43] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:44] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:45] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:45] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:46] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:46] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:47] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:48] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:49] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:50] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:51] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:51] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:51] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:52] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:53] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:00:53] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    }
   ],
   "source": [
    "app = dash.Dash()\n",
    "\n",
    "app.layout = html.Div([\n",
    "    dcc.Input(id='input', value='Enter Something Here!', type='text'),\n",
    "    html.Div(id='output')\n",
    "])\n",
    "\n",
    "@app.callback(\n",
    "    Output(component_id='output', component_property='children'),\n",
    "    [Input(component_id='input', component_property='value')]\n",
    ")\n",
    "def update_value(input_data):\n",
    "    try:\n",
    "        return str(float(input_data) ** 2)\n",
    "    except:\n",
    "        return \"Some Error\"\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    app.run_server()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>High</th>\n",
       "      <th>Low</th>\n",
       "      <th>Open</th>\n",
       "      <th>Close</th>\n",
       "      <th>Volume</th>\n",
       "      <th>Adj Close</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Date</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2014-12-31</th>\n",
       "      <td>45.136002</td>\n",
       "      <td>44.450001</td>\n",
       "      <td>44.618000</td>\n",
       "      <td>44.481998</td>\n",
       "      <td>11487500.0</td>\n",
       "      <td>44.481998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015-01-02</th>\n",
       "      <td>44.650002</td>\n",
       "      <td>42.652000</td>\n",
       "      <td>44.574001</td>\n",
       "      <td>43.862000</td>\n",
       "      <td>23822000.0</td>\n",
       "      <td>43.862000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015-01-05</th>\n",
       "      <td>43.299999</td>\n",
       "      <td>41.431999</td>\n",
       "      <td>42.910000</td>\n",
       "      <td>42.018002</td>\n",
       "      <td>26842500.0</td>\n",
       "      <td>42.018002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015-01-06</th>\n",
       "      <td>42.840000</td>\n",
       "      <td>40.841999</td>\n",
       "      <td>42.012001</td>\n",
       "      <td>42.256001</td>\n",
       "      <td>31309500.0</td>\n",
       "      <td>42.256001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2015-01-07</th>\n",
       "      <td>42.956001</td>\n",
       "      <td>41.956001</td>\n",
       "      <td>42.669998</td>\n",
       "      <td>42.189999</td>\n",
       "      <td>14842000.0</td>\n",
       "      <td>42.189999</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 High        Low       Open      Close      Volume  Adj Close\n",
       "Date                                                                         \n",
       "2014-12-31  45.136002  44.450001  44.618000  44.481998  11487500.0  44.481998\n",
       "2015-01-02  44.650002  42.652000  44.574001  43.862000  23822000.0  43.862000\n",
       "2015-01-05  43.299999  41.431999  42.910000  42.018002  26842500.0  42.018002\n",
       "2015-01-06  42.840000  40.841999  42.012001  42.256001  31309500.0  42.256001\n",
       "2015-01-07  42.956001  41.956001  42.669998  42.189999  14842000.0  42.189999"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import datetime\n",
    "import pandas_datareader.data as web\n",
    "\n",
    "start = datetime.datetime(2015, 1, 1)\n",
    "end = datetime.datetime.now()\n",
    "df = web.DataReader(\"TSLA\", 'yahoo', start, end)\n",
    "df.reset_index(inplace=True)\n",
    "df.set_index(\"Date\", inplace=True)\n",
    "#df = df.drop('Symbol', axis=1)\n",
    "\n",
    "df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      " * Serving Flask app \"__main__\" (lazy loading)\n",
      " * Environment: production\n",
      "   WARNING: This is a development server. Do not use it in a production deployment.\n",
      "   Use a production WSGI server instead.\n",
      " * Debug mode: off\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " * Running on http://127.0.0.1:8050/ (Press CTRL+C to quit)\n",
      "127.0.0.1 - - [30/Dec/2020 16:51:30] \"\u001b[37mGET / HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:51:32] \"\u001b[37mGET /_dash-dependencies HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 16:51:32] \"\u001b[37mGET /_dash-layout HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    }
   ],
   "source": [
    "app = dash.Dash()\n",
    "stock = 'TSLA'\n",
    "\n",
    "app.layout = html.Div(children=[\n",
    "    html.H1(children='whao, a graph!'),\n",
    "    \n",
    "    html.Div(children='Making a stock graph!'),\n",
    "    \n",
    "    dcc.Graph(\n",
    "        id='example-graph',\n",
    "        figure={\n",
    "            'data': [\n",
    "                {'x': df.index, 'y': df.Close, 'type': 'line', 'name': stock}\n",
    "            ],\n",
    "            'layout': {\n",
    "                'title': stock\n",
    "            }\n",
    "        }\n",
    "    )\n",
    "])\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    app.run_server()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      " * Serving Flask app \"__main__\" (lazy loading)\n",
      " * Environment: production\n",
      "   WARNING: This is a development server. Do not use it in a production deployment.\n",
      "   Use a production WSGI server instead.\n",
      " * Debug mode: off\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " * Running on http://127.0.0.1:8050/ (Press CTRL+C to quit)\n",
      "127.0.0.1 - - [30/Dec/2020 17:01:10] \"\u001b[37mGET / HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:01:10] \"\u001b[37mGET /_dash-layout HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:01:10] \"\u001b[37mGET /_dash-dependencies HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 153, in _read_one_data\n",
      "    resp = self._get_response(url, params=params)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 181, in _get_response\n",
      "    raise RemoteDataError(msg)\n",
      "pandas_datareader._utils.RemoteDataError: Unable to read URL: https://finance.yahoo.com/quote//history?period1=1420065000&period2=1609367399&interval=1d&frequency=1d&filter=history\n",
      "Response Text:\n",
      "b'<html>\\n<meta charset=\\'utf-8\\'>\\n<script>\\nvar u=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\';\\nif(window!=window.top){\\n  document.write(\\'<p>Content is currently unavailable.</p><img src=\"//geo.yahoo.com/p?s=1197757039&t=\\'+new Date().getTime()+\\'&_R=\\'+encodeURIComponent(document.referrer)+\\'&err=404&err_url=\\'+u+\\'\" width=\"0px\" height=\"0px\"/>\\');\\n}else{\\n  window.location.replace(u);\\n}\\n</script>\\n<noscript><META http-equiv=\"refresh\" content=\"0;URL=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\'\"></noscript>\\n</html>\\n'\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 153, in _read_one_data\n",
      "    resp = self._get_response(url, params=params)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 181, in _get_response\n",
      "    raise RemoteDataError(msg)\n",
      "pandas_datareader._utils.RemoteDataError: Unable to read URL: https://finance.yahoo.com/quote//history?period1=1420065000&period2=1609367399&interval=1d&frequency=1d&filter=history\n",
      "Response Text:\n",
      "b'<html>\\n<meta charset=\\'utf-8\\'>\\n<script>\\nvar u=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\';\\nif(window!=window.top){\\n  document.write(\\'<p>Content is currently unavailable.</p><img src=\"//geo.yahoo.com/p?s=1197757039&t=\\'+new Date().getTime()+\\'&_R=\\'+encodeURIComponent(document.referrer)+\\'&err=404&err_url=\\'+u+\\'\" width=\"0px\" height=\"0px\"/>\\');\\n}else{\\n  window.location.replace(u);\\n}\\n</script>\\n<noscript><META http-equiv=\"refresh\" content=\"0;URL=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\'\"></noscript>\\n</html>\\n'\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 153, in _read_one_data\n",
      "    resp = self._get_response(url, params=params)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 181, in _get_response\n",
      "    raise RemoteDataError(msg)\n",
      "pandas_datareader._utils.RemoteDataError: Unable to read URL: https://finance.yahoo.com/quote//history?period1=1420065000&period2=1609367399&interval=1d&frequency=1d&filter=history\n",
      "Response Text:\n",
      "b'<html>\\n<meta charset=\\'utf-8\\'>\\n<script>\\nvar u=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\';\\nif(window!=window.top){\\n  document.write(\\'<p>Content is currently unavailable.</p><img src=\"//geo.yahoo.com/p?s=1197757039&t=\\'+new Date().getTime()+\\'&_R=\\'+encodeURIComponent(document.referrer)+\\'&err=404&err_url=\\'+u+\\'\" width=\"0px\" height=\"0px\"/>\\');\\n}else{\\n  window.location.replace(u);\\n}\\n</script>\\n<noscript><META http-equiv=\"refresh\" content=\"0;URL=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\'\"></noscript>\\n</html>\\n'\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 153, in _read_one_data\n",
      "    resp = self._get_response(url, params=params)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 181, in _get_response\n",
      "    raise RemoteDataError(msg)\n",
      "pandas_datareader._utils.RemoteDataError: Unable to read URL: https://finance.yahoo.com/quote//history?period1=1420065000&period2=1609367399&interval=1d&frequency=1d&filter=history\n",
      "Response Text:\n",
      "b'<html>\\n<meta charset=\\'utf-8\\'>\\n<script>\\nvar u=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\';\\nif(window!=window.top){\\n  document.write(\\'<p>Content is currently unavailable.</p><img src=\"//geo.yahoo.com/p?s=1197757039&t=\\'+new Date().getTime()+\\'&_R=\\'+encodeURIComponent(document.referrer)+\\'&err=404&err_url=\\'+u+\\'\" width=\"0px\" height=\"0px\"/>\\');\\n}else{\\n  window.location.replace(u);\\n}\\n</script>\\n<noscript><META http-equiv=\"refresh\" content=\"0;URL=\\'https://www.yahoo.com/?err=404&err_url=https%3a%2f%2ffinance.yahoo.com%2fquote%2f%2fhistory%3fperiod1%3d1420065000%26period2%3d1609367399%26interval%3d1d%26frequency%3d1d%26filter%3dhistory\\'\"></noscript>\\n</html>\\n'\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "127.0.0.1 - - [30/Dec/2020 17:01:17] \"\u001b[1m\u001b[35mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 500 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:01:35] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:02:13] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:02:15] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:02:23] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aak using YahooDailyReader\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aak using YahooDailyReader\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aak using YahooDailyReader\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aak using YahooDailyReader\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "127.0.0.1 - - [30/Dec/2020 17:02:33] \"\u001b[1m\u001b[35mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 500 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:02:45] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:02:54] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aall using YahooDailyReader\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aall using YahooDailyReader\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aall using YahooDailyReader\n",
      "Exception on /_dash-update-component [POST]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 157, in _read_one_data\n",
      "    data = j[\"context\"][\"dispatcher\"][\"stores\"][\"HistoricalPriceStore\"]\n",
      "KeyError: 'HistoricalPriceStore'\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 2447, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1952, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1821, in handle_user_exception\n",
      "    reraise(exc_type, exc_value, tb)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\_compat.py\", line 39, in reraise\n",
      "    raise value\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1950, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\flask\\app.py\", line 1936, in dispatch_request\n",
      "    return self.view_functions[rule.endpoint](**req.view_args)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1076, in dispatch\n",
      "    response.set_data(func(*args, outputs_list=outputs_list))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\dash\\dash.py\", line 1007, in add_context\n",
      "    output_value = func(*args, **kwargs)  # %% callback invoked %%\n",
      "  File \"<ipython-input-12-3b4d4049aca1>\", line 16, in update_value\n",
      "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas\\util\\_decorators.py\", line 199, in wrapper\n",
      "    return func(*args, **kwargs)\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\data.py\", line 384, in DataReader\n",
      "    session=session,\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\base.py\", line 253, in read\n",
      "    df = self._read_one_data(self.url, params=self._get_params(self.symbols))\n",
      "  File \"C:\\Users\\Bhavesh kumar\\Anaconda3\\envs\\tensorflow_env\\lib\\site-packages\\pandas_datareader\\yahoo\\daily.py\", line 160, in _read_one_data\n",
      "    raise RemoteDataError(msg.format(symbol, self.__class__.__name__))\n",
      "pandas_datareader._utils.RemoteDataError: No data fetched for symbol aall using YahooDailyReader\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "127.0.0.1 - - [30/Dec/2020 17:03:07] \"\u001b[1m\u001b[35mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 500 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:03:15] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [30/Dec/2020 17:03:23] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    }
   ],
   "source": [
    "app = dash.Dash()\n",
    "\n",
    "app.layout = html.Div([\n",
    "    dcc.Input(id='input', value='', type='text'),\n",
    "    html.Div(id='output-graph')\n",
    "])\n",
    "\n",
    "@app.callback(\n",
    "    Output(component_id='output-graph', component_property='children'),\n",
    "    [Input(component_id='input', component_property='value')]\n",
    ")\n",
    "\n",
    "def update_value(input_data):\n",
    "    start = datetime.datetime(2015, 1, 1)\n",
    "    end = datetime.datetime.now()\n",
    "    df = web.DataReader(input_data, 'yahoo', start, end)\n",
    "    df.reset_index(inplace=True)\n",
    "    df.set_index(\"Date\", inplace=True)\n",
    "    \n",
    "    return dcc.Graph(\n",
    "        id='example-graph',\n",
    "        figure={\n",
    "            'data': [\n",
    "                {'x': df.index, 'y': df.Close, 'type': 'line', 'name': input_data},\n",
    "            ],\n",
    "            'layout': {\n",
    "                'title': input_data\n",
    "            }\n",
    "        }\n",
    "    )\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    app.run_server()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      "Dash is running on http://127.0.0.1:8050/\n",
      "\n",
      " * Serving Flask app \"__main__\" (lazy loading)\n",
      " * Environment: production\n",
      "   WARNING: This is a development server. Do not use it in a production deployment.\n",
      "   Use a production WSGI server instead.\n",
      " * Debug mode: off\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " * Running on http://127.0.0.1:8050/ (Press CTRL+C to quit)\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:28] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:28] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:28] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:30] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:30] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:31] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:32] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:33] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:34] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:35] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:36] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:37] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:38] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:39] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:40] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:41] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:42] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:43] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:44] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:45] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:46] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:47] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:48] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:49] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:50] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:51] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:52] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:53] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:54] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:55] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:56] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:57] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:58] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:14:59] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:00] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:01] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:02] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:03] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:04] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:05] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:06] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:07] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:08] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:09] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:10] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:11] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:12] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:13] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:14] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:15] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:16] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:17] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:18] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:19] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:20] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:21] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:22] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:23] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:24] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:25] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:26] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:27] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:28] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:29] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:30] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:31] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:32] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:33] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:34] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:35] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:36] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:37] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:38] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:39] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:40] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:41] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:42] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:43] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:44] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:45] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:46] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:47] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:48] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:49] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:50] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:51] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:52] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:53] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:54] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "127.0.0.1 - - [31/Dec/2020 11:15:55] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:56] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:57] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:58] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:15:59] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:00] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:01] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:02] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:03] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:04] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:05] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:06] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:07] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:08] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:11] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:11] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:13] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:13] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:15] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n",
      "127.0.0.1 - - [31/Dec/2020 11:16:15] \"\u001b[37mPOST /_dash-update-component HTTP/1.1\u001b[0m\" 200 -\n"
     ]
    }
   ],
   "source": [
    "import dash\n",
    "from dash.dependencies import Output, Input\n",
    "import dash_core_components as dcc\n",
    "import dash_html_components as html\n",
    "import plotly.graph_objs as go\n",
    "import random\n",
    "from collections import deque\n",
    "\n",
    "X = deque(maxlen=20)\n",
    "X.append(1)\n",
    "Y = deque(maxlen=20)\n",
    "Y.append(1)\n",
    "\n",
    "app = dash.Dash(__name__)\n",
    "app.layout = html.Div([\n",
    "    dcc.Graph(id='live-graph', animate=True),\n",
    "    dcc.Interval(id='graph-update', interval = 1*1000),\n",
    "])\n",
    "\n",
    "@app.callback(Output('live-graph', 'figure'),\n",
    "             [Input('graph-update', 'n_intervals')])\n",
    "\n",
    "def update_graph_scatter(input_data):\n",
    "    X.append(X[-1] + 1)\n",
    "    Y.append(Y[-1] + Y[-1] * random.uniform(-0.1, 0.1))\n",
    "    \n",
    "    data = go.Scatter(x=list(X), y=list(Y), name='Scatter',\n",
    "                     mode='lines+markers')\n",
    "    \n",
    "    return {'data': [data], 'layout': go.Layout(xaxis=dict(range=[min(X), max(X)]),\n",
    "                                               yaxis=dict(range=[min(Y), max(Y)]))}\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    app.run_server()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
